enterprise/reporting/reporting.go

```diff
@@ -400,7 +415,12 @@ func (r *DefaultReporter) mainLoop(ctx context.Context, c types.SyncerConfig) {
 			currentMin := time.Now().UTC().Unix() / 60
 
 			getReportsStart := time.Now()
-			reports, reportedAt, err := r.getReports(currentMin, c.ConnInfo)
+			aggregationIntervalMin := int64(aggregationInterval.Load().Minutes())
+			if aggregationIntervalMin <= 0 || 60%aggregationIntervalMin != 0 {
+				panic(fmt.Errorf("[ Reporting ]: Error aggregationIntervalMinutes should be a factor of 60, got %d", aggregationIntervalMin))
```

Comment:
this panic will cause application to halt processing until we fix the config which may take time during code freeze, do you think instead of panic we should add an error log and raise a log based alert for this?

Also please specify why are we adding this check in comments

Reply:
One potential issue with continuing the processing is that data from the next hour might get aggregated with the current hour’s data. This is undesirable and needs to be avoided.

Reply:
In that case, fire an alert and don't initialize reporter. Processor's main task should be to process the events. Reporting is the secondary task which should not block the primary task.